docs/source/cpp/csv.rst

```diff
@@ -25,15 +25,26 @@ Reading and Writing CSV files
 =============================
 
 Arrow provides a fast CSV reader allowing ingestion of external data
-as Arrow tables.
+as Arrow Tables or streamed as Arrow RecordBatches.
 
 .. seealso::
    :ref:`CSV reader/writer API reference <cpp-api-csv>`.
 
-Basic usage
-===========
+Reading CSV files
+=================
+
+Data in a CSV file can either be read in as a single Arrow Table using
+:class:`~arrow::csv::TableReader` or streamed as RecordBatches using
+:class:`~arrow::csv::StreamingReader`. See :ref:`Tradeoffs <cpp-csv-tradeoffs>` for a
+discussion of the tradeoffs between the two methods.
+
+Both these readers require an :class:`arrow::io::InputStream` instance
```

Comment:
Not a problem at the moment.  However, there is [a PR](https://github.com/apache/arrow/pull/14269) which would allow you to get slightly better performance if you use a random access file (typically if you're connected to cloud storage) instead of an input stream.  So we might need to update this text after that pull merges.

Reply:
Thanks for the heads up. I'll keep track of that PR.